### PR TITLE
Ensure IOperation/CFG doesn't crash in presence of using declarations:

### PR DESCRIPTION
**Pull request by chsienki**
- Make using declarations return operationkind.none
- Return the declarations as child nodes of the using declaration
- Add a couple of placeholder tests to ensure nothing crashes

### CODE REVIEW

**Comment by chsienki** [comment]
@dotnet/roslyn-compiler for review please

**Comment by AlekseyTs** [comment]
While this PR is under review, please make sure that running all compiler tests with IOperation hook enabled doesn't encounter failures due to the Using Declarations feature.

File: src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IUsingStatement.cs [hunk 1]
```diff
@@ -2931,5 +2932,82 @@ void M()
...
+        {
+            string source = @"
+
+using System;
+class P : IDisposable
+{
+    void M()
+    /*<bind>*/{
+        System.Console.WriteLine(""before"");
+        using var x = new P();
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IUsingStatement.cs]: >using var x = new P(); [](start = 8, length = 22)

Please add tests:
- with multiple locals declared by the same statement. 
- with a control flow in the first initializer
- with a control flow in the second initializer

File: src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IVariableDeclaration.cs [hunk 2]
```diff
@@ -1498,6 +1498,48 @@ static void Main(string[] args)
...
+
+class C : IDisposable
+{
+    public void Dispose()
+    {
+    }
+
+    public static void M1()
+    {
+        /*<bind>*/using var c = new C();/*</bind>*/
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IVariableDeclaration.cs]: > /*<bind>*/using var c = new C();/*</bind>*/ [](start = 7, length = 44)

Please add a test for multiple declarations. Also, it feels like this test belongs in the file dedicated to ```using``` statements

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 1)

File: src/Compilers/CSharp/Portable/BoundTree/Statement.cs [hunk 3]
```diff
@@ -30,4 +30,9 @@ partial class BoundPointerIndirectionOperator
     {
         protected override ImmutableArray<BoundNode> Children => ImmutableArray.Create<BoundNode>(this.Operand);
     }
+
+    partial class BoundUsingLocalDeclarations
+    {
+        protected override ImmutableArray<BoundNode> Children => this.LocalDeclarations.AsImmutable<BoundNode>();
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/BoundTree/Statement.cs]: Use [http://source.roslyn.io/#Microsoft.CodeAnalysis/Collections/StaticCast.cs,97d1e4a8dd39ade0,references](http://source.roslyn.io/#Microsoft.CodeAnalysis/Collections/StaticCast.cs,97d1e4a8dd39ade0,references), which will staticly verify the cast is safe at compile time and won't cause the existing immutable array to be boxed into an ienumerable.

File: src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IUsingStatement.cs [hunk 4]
```diff
@@ -2931,5 +2932,82 @@ void M()
...
+        public void UsingDeclaration_Flow()
+        {
+            string source = @"
+
+using System;
+class P : IDisposable
+{
+    void M()
+    /*<bind>*/{
+        System.Console.WriteLine(""before"");
```
Comment (333fred) [inline src/Compilers/CSharp/Test/IOperation/IOperation/IOperationTests_IUsingStatement.cs]: Please remove the before and after printlines, they add a bunch of lines to the diff and aren't really adding any test info.

**Comment by 333fred** [comment]
Done review pass (commit 1)

File: src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs [hunk 5]
```diff
@@ -2119,5 +2123,16 @@ private IOperation CreateRangeExpressionOperation(BoundRangeExpression boundRang
                 rightOperand: new Lazy<IOperation>(() => Create(boundRange.RightOperand)),
                 symbol: boundRange.MethodOpt);
         }
+        private IOperation CreateUsingLocalDeclarationsOperation(BoundUsingLocalDeclarations boundNode)
+        {
+            //TODO: Implement UsingLocalDeclaration operations correctly.
+            //      For now we return an implicit operationNone, with a single child consisting of the using declaration parsed as if it were a standard variable declaration
```
Comment (333fred) [inline src/Compilers/CSharp/Portable/Operations/CSharpOperationFactory.cs]: >implicit [](start = 41, length = 8)

This operation is not implicit.

**Pull request merged**